### ISSUE TITLE
Chrome giving 'TypeError: Cannot read property 'length' of null'

### ISSUE THREAD

**Issue by wooly** [exupero/saveSvgAsPng#7]
Hey,

This plugin looks awesome, but unfortunately I've not been able to get it working.

I have the following code generated by d3:

``` html
<div id="breakdown-chart">
  <svg id="fish" width="340" height="340">
    <g id="pieChart" transform="translate(170,170)">
      <path fill="#d5dadc" d="M9.797174393178826e-15,-160A160,160 0 0,1 125.09303719488477,-99.75836829739735L53.16454080782603,-42.397306526393876A68,68 0 0,0 4.1637991171010006e-15,-68Z">
      </path>
      <path fill="#d64e40" d="M125.09303719488477,-99.75836829739735A160,160 0 0,1 125.09303719488477,-99.75836829739735L53.16454080782603,-42.397306526393876A68,68 0 0,0 53.16454080782603,-42.397306526393876Z">
      </path>
      <path fill="#f78837" d="M125.09303719488477,-99.75836829739735A160,160 0 0,1 125.09303719488477,-99.75836829739735L53.16454080782603,-42.397306526393876A68,68 0 0,0 53.16454080782603,-42.397306526393876Z">
      </path>
      <path fill="#ecc837" d="M125.09303719488477,-99.75836829739735A160,160 0 0,1 125.09303719488477,-99.75836829739735L53.16454080782603,-42.397306526393876A68,68 0 0,0 53.16454080782603,-42.397306526393876Z">
      </path>
      <path fill="#8bd38b" d="M125.09303719488477,-99.75836829739735A160,160 0 1,1 -2.9391523179536473e-14,-160L-1.2491397351303002e-14,-68A68,68 0 1,0 53.16454080782603,-42.397306526393876Z">
      </path>
      <path fill="#43b373" d="M-2.9391523179536473e-14,-160A160,160 0 0,1 -2.9391523179536473e-14,-160L-1.2491397351303002e-14,-68A68,68 0 0,0 -1.2491397351303002e-14,-68Z">
      </path>
    </g>
  </svg>
</div>
```

But when I try to save as a PNG in console, I get the TypeError. The command I type in the Chrome console is `saveSvgAsPng(document.getElementById('fish'), 'filename.png')`

Any ideas?

Thanks!

**Comment by exupero** [comment]
Are you using the latest version of the script? I encountered the same problem a few days ago and pushed a fix.

**Comment by wooly** [comment]
Yep, I cloned the repo just today and copied the js file across.

**Comment by exupero** [comment]
Does the latest version of the code produce this error?

**Comment by lobo-tuerto** [comment]
I'm getting an error too, maybe it's related. I just got the file from the master branch and I'm seeing this error:

```
Error: Failed to execute 'querySelectorAll' on 'Element': '[ng:cloak], [ng-cloak], [data-ng-cloak], [x-ng-cloak], .ng-cloak, .x-ng-cloak, .ng-hide:not(.ng-animate)' is not a valid selector.
```

Tried this on a svg generated through a directive in angular (in case you were wondering about where does those names come from).

Invoking line:

```
saveSvgAsPng(document.getElementById("diagram"), "diagram.png", 3);
```

**Comment by exupero** [comment]
@lobo-tuerto I don't think this is related. The problem is that `document.querySelectorAll` doesn't recognize `[ng:cloak]`, which is used in a CSS rule.

There's probably a better way to handle styles than how the script currently does it. Please create a separate issue and I'll take a closer look when I have a chance.

**Comment by lobo-tuerto** [comment]
Done. Thank you.

**Comment by exupero** [comment]
I'm closing this since I haven't heard any more about the original problem. If anyone sees `TypeError: Cannot read property 'length' of null`, feel free to re-open.

**Comment by honi** [comment]
I'm getting this error. In my case, it is line 43:
https://github.com/exupero/saveSvgAsPng/blob/gh-pages/saveSvgAsPng.js#L43

The stylesheet that has `cssRules == null` in my case is a Google Font. So maybe a simple check `if (rules) ...` before the for loop could fix this.

**Comment by trhall** [comment]
I ran into a similar issue as @honi above, but with a Google Chart. While Google Charts has the ability to [create a PNG](https://developers.google.com/chart/interactive/docs/printing), that function only works for their core charts. I have a Sankey chart generated with Google Charts that gave me the same "Cannot read property 'length' of null" error. Didn't have the same issue while creating a Sankey using D3.js.

Just sent a pull request with my suggested patch. To replicate, I used a sample Sankey chart from the code on the Google Charts [Sankey page](https://developers.google.com/chart/interactive/docs/gallery/sankey). Adding the null check resolves the issue. I wonder if Google has a similar cssRules == null, as @honi mentions?
